umbraco/Umbraco-CMS
# fix: #12253 Anchor picker does not appear

This relates to #12253 

### Steps to test 

1. Install Umbraco v11 
2. Use database attached [Umbraco.sqlite.zip](https://github.com/umbraco/Umbraco-CMS/files/10105668/Umbraco.sqlite.zip)
3. Open umbraco with u/n : marcoteodoro@double.pt p/w: 0123456789
4. Open destination node and create a new anchor link from the origin node. (you will see that anchor links from the blocklist editor and grid editor are listed on the dropdown)

![chrome-capture-2022-10-28](https://user-images.githubusercontent.com/3866444/204335189-2a25d2cf-e7b3-4e55-bfea-5af52c9dc8a8.gif)


### Description

`GetAnchorValuesFromRTEs` method was updated to include `BlockList` and `BlockGrid`
https://github.com/umbraco/Umbraco-CMS/blob/c9410391bf5cc7a05c138208fbbbb404e85847bf/src/Umbraco.Core/Services/ContentServiceExtensions.cs#L80-L80

on top of this, i had to include this line of code `var rteUnescaped = Regex.Unescape(rteContent);` to make the regex work. https://github.com/umbraco/Umbraco-CMS/blob/c9410391bf5cc7a05c138208fbbbb404e85847bf/src/Umbraco.Core/Services/ContentServiceExtensions.cs#L103-L103

However, one thing i noticed is that this that when the culture variable is assigned with `*` this method is always returning null contentProperty.GetValue(culture)?.ToString();`

https://github.com/umbraco/Umbraco-CMS/blob/c9410391bf5cc7a05c138208fbbbb404e85847bf/src/Umbraco.Core/Services/ContentServiceExtensions.cs#L73

So this PR is just an initial commit to see what others can suggest. Is this last issue a known issue? or I'm not using it correctly?
